proposals/0007-access-policy.md

```diff
@@ -0,0 +1,126 @@
+---
+simd: '0007'
+title: SIMD Access Policy
+authors:
+- Jacob Creech (Solana Foundation)
+category: Meta
+type: Meta
+status: Draft
+created: 2022-12-10
+---
+
+## Summary
+
+3 levels of access, in the order of increasing access:
+
+1. Triage
+
+    Requirement: One voucher from anyone with level 2 or above access.
+
+2. Core
+
+    Requirement: One voucher from anyone with level 3 access or two from
+    anyone with level 2 access.
+
+3. Maintain
+
+    Requirement: One voucher from anyone with level 3 access. This permission
+    is usually reserved for those maintaining the SIMD process.
+
+Each of the levels implies having the previous levels - e.g. level 2 implies
+level 1. A Core Contributor will be able to triage pull requests in accordance
+with Github access policies.
+
+## Motivation
+
+Shamelessly borrowing from [Mozilla's access
+policy](https://www.mozilla.org/en-US/about/governance/policies/commit/access-policy/): 
+
+```
+There are two sorts of control which can be used to stop people checking in -
+technical and social.
+
+A "full technical" implementation would have per-directory permissions
+everywhere, but would lead to a greatly-increased management overhead for IT,
+vouchers and developers alike.
+A "full social" implementation would just have a single permission which gave
+you complete access to everything, but (depending on the height of the barrier
+to that permission) there is a risk of making developer's lives more difficult
+when they are excluded, or of giving the untrustworthy or incompetent power to
+mess things up.
+Therefore, a good policy balances the use of technical and social controls to
+minimize both management overhead and risk to the development process.
+```
+
+## Alternatives Considered
+
+[EIP](https://github.com/ethereum/EIPs) run under two separate levels,
+contributors and editors. Editors are a select group of people that review each
+EIP and have write access. Instead of starting with a select group of editors
+who have exclusive write access, we could begin with an easier way to achieve
+write access and gradually tighten permissions as needed.
+
+[RFCs](https://www.rust-lang.org/governance) have sub-teams that review and
+triage new proposed changes. This is a possibility, but would require more
+granular labeling of different SIMDs.
+
+[PEP](https://peps.python.org/pep-0013/) has an election of a council that
+reviews and triages new proposed changes. The PEP approach may result in an
+increase in governance overhead.
+
+## New Terminology
+
+Core Contributor - Individuals with Core access to SIMDs that can assist in
+triaging different SIMDs.
+
+## Detailed Design
+
+### Level 1 - Triage Access
```

Comment:
How were you thinking of actually implementing the vouching process? Again stealing from Mozilla, they do this by way of bug tracker:
1. The person requesting access opens a new issue, with their specific request
2. Voucher does so by commenting on the issue
3. Devops then sees the vouch and implements the access change

Removes are documented in a similar way. 

This generates a really nice audit log for access changes

Reply:
I was going to do exactly this. Build a git bot that watches issues for vouchers and maintains the addition/removal of access.

Reply:
Using humans seems fair enough to start with too, building a bot seems great until it becomes a bigger burden than a meat-powered process. Just my immediate thought on that

Reply:
That seems fine for now tbh, a small list :)
Your choice of course, just sharing what I'd do to start with. I find it hard to imagine in the short term there'll be an unmanageable flow of requests for access. I'd like to be proven wrong though!

Reply:
Updated with Access request process, also stolen from Mozilla.